common/src/main/java/com/microsoft/identity/common/internal/ui/webview/challengehandlers/YubiKitUsbSmartcardCertBasedAuthManager.java

```diff
@@ -0,0 +1,204 @@
+import com.yubico.yubikit.android.transport.usb.UsbYubiKeyManager;
+import com.yubico.yubikit.android.transport.usb.connection.UsbSmartCardConnection;
+import com.yubico.yubikit.core.smartcard.SmartCardConnection;
+import com.yubico.yubikit.core.util.Callback;
+import com.yubico.yubikit.core.util.Result;
+import com.yubico.yubikit.piv.PivSession;
+
+import java.io.IOException;
+import java.security.PrivateKey;
+import java.security.cert.X509Certificate;
+import java.util.concurrent.Callable;
+
+public class YubiKitUsbSmartcardCertBasedAuthManager extends AbstractUsbSmartcardCertBasedAuthManager {
+    private static final String TAG = YubiKitUsbSmartcardCertBasedAuthManager.class.getSimpleName();
+    private static final String DEVICE_ERROR_MESSAGE = "No USB device is currently connected.";
+
+    private final UsbYubiKeyManager mUsbYubiKeyManager;
+    private UsbYubiKeyDevice mUsbDevice;
+
+    //Lock to help facilitate synchronization
+    private static final Object sDeviceLock = new Object();
+
+    public YubiKitUsbSmartcardCertBasedAuthManager(@NonNull final Context context) {
+        mUsbYubiKeyManager = new UsbYubiKeyManager(context.getApplicationContext());
+    }
+
+    /**
+     * Create and start YubiKitManager for UsbDiscovery mode.
+     * When in Usb Discovery mode, Yubikeys that plug into the device will be accessible
+     *  once the user provides permission via the Android permission dialog.
+     */
+    @Override
+    boolean startDiscovery(@NonNull final Activity activity) {
+        mUsbYubiKeyManager.enable(new UsbConfiguration(), new Callback<UsbYubiKeyDevice>() {
+            @Override
+            public void invoke(@NonNull UsbYubiKeyDevice device) {
+                Logger.verbose(TAG, "A YubiKey device was connected via usb.");
+                synchronized (sDeviceLock) {
+                    mUsbDevice = device;
+                    if (mConnectionCallback != null) {
+                        mConnectionCallback.onCreateConnection();
+                    }
+
+                    mUsbDevice.setOnClosed(new Runnable() {
+                        @Override
+                        public void run() {
+                            Logger.verbose(TAG, "A YubiKey device was disconnected via usb.");
+                            synchronized (sDeviceLock) {
+                                mUsbDevice = null;
+                            }
+                            YubiKeyPivProviderManager.removePivProvider();
+                            if (mConnectionCallback != null) {
+                                mConnectionCallback.onClosedConnection();
+                            }
+                        }
+
+                    });
+                }
+            }
+        });
+        return true;
+    }
+
+    /**
+     * Ceases usb discovery for YubiKeys.
+     * Usually called when a host fragment is destroyed.
+     */
+    @Override
+    void stopDiscovery(@NonNull final Activity activity) {
```

Comment:
note: let's add a comment/details on how usb's stopDiscovery() is inconsistent (for future devs to understand why stopDiscovery()'s invoking pattern is not consistent between the two flows).